src/main/java/org/broadinstitute/hellbender/utils/codecs/refseq/RefSeqFeature.java

```diff
@@ -0,0 +1,236 @@
+//            }
+//        }
+//
+//        return relevantExons;
+//    }
+//
+//    /** convenience method; returns the numbers of the exons in the interval **/
+//    public List<Integer> getExonNumbersInInterval( Locatable interval ) {
+//        List<Integer> numbers = new ArrayList<Integer>();
+//        int iNo = 0;
+//        for ( Locatable exon : getExons() ) {
+//            if ( IntervalUtils.containsP(interval, exon) ) {
+//                numbers.add(iNo);
+//            }
+//            iNo++;
+//        }
+//
+//        return numbers;
+//    }
+
+    public String getTranscriptUniqueGeneName() {
+        return String.format("%s(%s)",getGeneName(),getTranscriptId());
+    }
+
+//    ArrayList<GenomeLoc> exonInRefOrderCache = null;
+
+//    public Integer getSortedOverlapInteger(GenomeLoc position) {
+//        int exonNo = -1;
+//        ArrayList<GenomeLoc> exonsInReferenceOrder = exonInRefOrderCache != null ? exonInRefOrderCache : new ArrayList<GenomeLoc>(exons);
+//        if ( exonInRefOrderCache == null ) {
+//            Collections.sort(exonsInReferenceOrder);
+//        }
+//        exonInRefOrderCache = exonsInReferenceOrder;
+//        for ( GenomeLoc exon : exonsInReferenceOrder ) {
+//            if ( exon.overlapsP(position) ) {
+//                return ++exonNo;
+//            }
+//            ++exonNo;
+//        }
+//
+//        return -1;
+//    }
+
+//    public GenomeLoc getSortedExonLoc(int offset) {
+//        ArrayList<GenomeLoc> exonsInReferenceOrder = exonInRefOrderCache != null ? exonInRefOrderCache : new ArrayList<GenomeLoc>(exons);
+//        if ( exonInRefOrderCache == null ) {
+//            Collections.sort(exonsInReferenceOrder);
+//        }
+//        exonInRefOrderCache = exonsInReferenceOrder;
+//        return exonsInReferenceOrder.get(offset);
+//    }
+
+//    /** Returns true if the specified interval 'that' overlaps with the full genomic interval of this transcript */
+//    public boolean overlapsP (Locatable that) {
+//        return IntervalUtils.overlaps(this, that);
+//    }
+//
+//    /** Returns true if the specified interval 'that' overlaps with the coding genomic interval of this transcript.
+//     * NOTE: since "coding interval" is still a single genomic interval, it will not contain UTRs of the outermost exons,
+//     * but it will still contain introns and/or exons internal to this genomic locus that are not spliced into this transcript.
+//     * @see #overlapsExonP
+//     */
+//    public boolean overlapsCodingP (Locatable that) {
+//        return transcript_coding_interval.overlapsP(that);
+//    }
+//
+    /**
+     * Returns a count of the total number of reference bases spanned by gene summary. Will total the length of the exons or
+     * if absent, the lengthOnReference for the gene itself.
+     *
+     * NOTE: This currently makes the assumption that genes do not ever have overlapping exons. I do not know if this is a fair
```

Comment:
According to my sources, genes **can** have overlapping exons, though...

Reply:
I say lets burn this bridge when we get to it. It is a fairly edge case functionality that GATK3 used and ideally we will replace this class entirely with something more general